app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,56 @@
+class PostsController < ApplicationController
+  before_action :find_user, only: %i[index show like unlike]
+  before_action :find_post, only: %i[show like unlike]
+
+  def index
+    @posts = @user.posts
+  end
+
+  def show; end
+
+  def new
+    @user = current_user
+    @post = @user.posts.new
+  end
+
+  def create
+    @post = current_user.posts.new(post_params)
+    if @post.save
+      flash[:notice] = 'Post created successfully.'
+      redirect_to user_path(current_user)
+    else
+      render 'new'
+    end
+  end
+
+  def like
+    @like = @post.likes.new
+    @like.author = current_user
+    @like.save
+    redirect_to user_post_path(@user, @post)
+  end
+
+  def unlike
+    @like = @post.likes.find_by(post: @post) # Find the like
+    @like&.destroy # Destroy the like if found
+    redirect_to user_post_path(@user, @post)
+  end
```

Comment:
- [ ] Good job implementing the like functionality, however, this logic should be implemented in its own controller file. Kindly create a `likes_controller.rb` file in the `/controllers` folder and implement the logic there.

#### If need be also update your Like and unlike routes so that the application does not break